py/server/tests/test_partitioned_table_service.py

```diff
@@ -0,0 +1,248 @@
+        self.partitions_size_subscriptions: dict[PartitionedTableLocationKey, bool] = {}
+        self.existing_partitions_called = 0
+        self.partition_size_called = 0
+
+    def table_schema(self, table_key: TableKey) -> Tuple[pa.Schema, Optional[pa.Schema]]:
+        if table_key.key == "test":
+             return self.pt_schema, self.pc_schema
+        return pa.Schema(), None
+
+    def existing_partitions(self, table_key: TableKey, callback: Callable[[PartitionedTableLocationKey, Optional[pa.Table]], None]) -> None:
+        pa_table = next(self.gen_pa_table)
+        if table_key.key == "test":
+            ticker = str(pa_table.column("Ticker")[0])
+
+            partition_key = PartitionedTableLocationKey(f"{ticker}/NYSE")
+            self.partitions[partition_key] = pa_table
+
+            expr = ((pc.field("Ticker") == f"{ticker}") & (pc.field("Exchange") == "NYSE"))
+            callback(partition_key, pa_table.filter(expr).select(["Ticker", "Exchange"]).slice(0, 1))
+            self.existing_partitions_called += 1
+
+    def partition_size(self, table_key: TableKey, table_location_key: PartitionedTableLocationKey,
+                       callback: Callable[[int], None]) -> None:
+        callback(self.partitions[table_location_key].num_rows)
+        self.partition_size_called += 1
+
+    def column_values(self, table_key: TableKey, table_location_key: PartitionedTableLocationKey,
+                      col: str, offset: int, min_rows: int, max_rows: int) -> pa.Table:
+        if table_key.key == "test":
+            return self.partitions[table_location_key].select([col]).slice(offset, max_rows)
+        else:
+            return pa.table([])
+
+    def _th_new_partitions(self, table_key: TableKey, exec_ctx: ExecutionContext, callback: Callable[[PartitionedTableLocationKey, Optional[pa.Table]], None]) -> None:
+        if table_key.key != "test":
+            return
+
+        while not self.sub_new_partition_cancelled:
+            try:
+                with exec_ctx:
+                    pa_table = next(self.gen_pa_table)
+            except StopIteration:
+                break
+
+            ticker = str(pa_table.column("Ticker")[0])
+            partition_key = PartitionedTableLocationKey(f"{ticker}/NYSE")
+            self.partitions[partition_key] = pa_table
+
+            expr = ((pc.field("Ticker") == f"{ticker}") & (pc.field("Exchange") == "NYSE"))
+            callback(partition_key, pa_table.filter(expr).select(["Ticker", "Exchange"]).slice(0, 1))
+            time.sleep(0.1)
+
+    def subscribe_to_new_partitions(self, table_key: TableKey, callback) -> Callable[[], None]:
+        if table_key.key != "test":
+            return lambda: None
+
+        exec_ctx = get_exec_ctx()
+        th = threading.Thread(target=self._th_new_partitions, args=(table_key, exec_ctx, callback))
+        th.start()
+
+        def _cancellation_callback():
+            self.sub_new_partition_cancelled += 1
```

Comment:
Is this thread safe in 3.13t?  The value is being accessed from two different threads without protection.